docs/DafnyRef/UserGuide.md

```diff
@@ -51,129 +96,315 @@ verification failure in a file that is being checked.
-the `--input` option for each extra file or use `dafny build` to build the executable and then run it in an 
-additional step.
 
-**Note:** `dafny run` will typically produce the same results as the executables produced by `dafny build`.  The only expected differences are performance (`dafny run` may not optimize as much as `dafny build`), and target-language-specific configuration issues (e.g. encoding issues: `dafny run` sets language-specific flags to request UTF-8 output for the [`print`](#print-encoding) statement in all languages, whereas `dafny build` leaves language-specific runtime configuration to the user).
 
-The command-line also expects the following:
-- Files are designated by absolute paths or paths relative to the current
-working directory. A command-line argument not matching a known option is considered a filepath, and likely one
-with an unsupported suffix, provoking an error message..
-- Files containing dafny code must have a `.dfy` suffix.
-- There must be at least one `.dfy` file.
-- The command-line may contain other kinds of files appropriate to
-the language that the Dafny files are being compiled to.
-- The option `--` means that all subsequent command-line arguments are not options to the dafny tool; they are either files or arguments to the `dafny run` command.
-- Old-style options may begin with either a `/` (as is typical on Windows) or a `-` (as is typical on Linux)
-- If an option is repeated (e.g., with a different argument), then the later instance on the command-line supersedes the earlier instance.
-- If an option takes an argument, the option name is followed by a `:` (old-style) or `=` (new-style) and then by the argument value, with no
-intervening white space; if the argument itself contains white space, the argument must be enclosed in quotes.
-- Escape characters are determined by the shell executing the command-line.
+**Note:** `dafny run` will typically produce the same results as the executables produced by `dafny build`.  The only expected differences are these:
+- performance --- `dafny run` may not optimize as much as `dafny build`
+- target-language-specific configuration issues ---  e.g. encoding issues: `dafny run` sets language-specific flags to request UTF-8 output for the [`print`](#print-encoding) statement in all languages, whereas `dafny build` leaves language-specific runtime configuration to the user.
+
+#### 25.5.1.7. `dafny server` {#sec-dafny-server}
+
+The `dafny server` command starts the Dafny Language Server, which as an [LSP-compliant](https://microsoft.github.io/language-server-protocol/) implementation of Dafny.
+Note two important points:
+- The [Dafny VSCode extension]() uses this LSP implementation
+- This LSP implementation is designed so that its behavior (other than IDE specific aspects) matches that of the command-line `dafny` tool
```

Comment:
> This LSP implementation is designed so that its behavior (other than IDE specific aspects) matches that of the command-line `dafny` tool

I'm not sure what this means except that the language server implements the same Dafny semantics as the other commands, and that some of the options are shared, but that's what I would expect.

Reply:
OK. Revised the text a bit, but still I think the reassurance to the users that they will get the same results and semantics either way is OK to say.